### ISSUE TITLE
[SR-11943] Outdated documentation for `numericCast(_:)`

### ISSUE THREAD

**Issue by swift-ci** [apple/swift#54363]
|                  |                 |
   |------------------|-----------------|
   |Previous ID       | SR-11943      |
   |Radar             | rdar://problem/57888487         |
   |Original Reporter | bzamayo (JIRA User)      |
   |Type              | Improvement    |

   <details>
  <summary>Additional Detail from JIRA</summary>

   |                  |                 |
   |------------------|-----------------|
   |Votes             | 0         |
   |Component/s       | Standard Library    |
   |Labels            | Improvement, StarterBug, StarterDocs        |
   |Assignee          | @natecook1000      |
   |Priority          | Medium      |

   

   md5: e13604a2bc1f0808504231367a9bedab

  </details>

**Issue Description:**

The [documentation](https://developer.apple.com/documentation/swift/2884564-numericcast) for this standard library could be improved in a few ways. Discussion of modern constructor syntax for type conversions should be included (like \`UInt32(value)\` or \`Int(value)\` or whatever), as generally that is the preferred Swift language idiom over calling \`numericCast\`.

However, my main reason for filing the bug is the example code is outdated:

 

``` java
func random(in range: Range<Int>) -> Int {
    return numericCast(arc4random_uniform(numericCast(range.count)))
        + range.lowerBound
}
```

 

It uses \`numericCast\` twice in the context of creating a random number generation function. Clearly, this example was written before the standard library incorporated native random APIs. This example should be changed to something outside of the random number domain, as inventing \``` random(in🙂` `` functions ad-hoc (and especially calls to \`arc4random_uniform\`) from Swift should no longer be encouraged as good practice.

**Comment by ArielDemarco** [comment]
Hi @beccadax! This seems to be already fixed [here](https://github.com/apple/swift/pull/31464/)

**Comment by beccadax** [comment]
Yup. Fixed by #31464.
